service/controller/v2/resource/deployment/resource.go

```diff
@@ -147,3 +194,22 @@ func (r *Resource) getDeploymentsClient() (*azureresource.DeploymentsClient, err
 
 	return azureClients.DeploymentsClient, nil
 }
+
+func (r *Resource) getVMsClient() (*compute.VirtualMachineScaleSetVMsClient, error) {
```

Comment:
One idea I had was that we could have these methods being assigned to `r.azureConfig`. We have the config everywhere and the methods here are repeated all over the repo. This could be nicer.